packages/dom/src/dom/is-number-input.js

```diff
@@ -2,12 +2,16 @@
  * Check whether the given element is an input field of type number
  * and has a valueAsNumber
  *
- * @param {HTMLElement} element The HTML element.
+ * @param {Element} element The HTML element.
  *
  * @return {boolean} True if the element is input and holds a number.
  */
 export default function isNumberInput( element ) {
-	const { nodeName, type, valueAsNumber } = element;
+	const {
+		nodeName,
+		type,
+		valueAsNumber,
+	} = /** @type {HTMLInputElement} */ ( element );
 
 	return nodeName === 'INPUT' && type === 'number' && !! valueAsNumber;
```

Comment:
Our `isHTMLInputElement` function seems perfect for this:
```suggestion
	return isHTMLInputElement( element ) && !! element.valueAsNumber;
```